src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -35,8 +35,8 @@
 
 #ifdef __GNUC__
 
-// __nop needs volatile so that compiler doesn't optimize it away
-#define NOP() asm volatile ("nop");
+// ISO C++ asm is always implicitly volatile
```

Comment:
I can find no evidence for this claim, and it seems to me likely incorrect.  This is also way outside the
described scope of this PR.

Reply:
Hi Kim, it's actually listed under https://gcc.gnu.org/onlinedocs/gcc/Basic-Asm.html as follows:

> Qualifiers
volatile
The optional volatile qualifier has no effect. All basic asm blocks are implicitly volatile.

I'll take this outside of this PR though